### PR TITLE
Add/log event module

### LINKED ISSUE #647: Implement the `logEvent` function in api2

### PR DESCRIPTION
**Pull request by saucecodee**
Fixes #647

- Create and abstract shared module
- Add `EventLog` module 
- Implement  `logEvent` function

### CODE REVIEW

File: packages/api2/src/event-log/event-log.controller.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Controller, Get, Param } from '@nestjs/common';
+import { EventLogService } from './event-log.service';
+
+@Controller('event-log')
+export class EventLogController {
```
Comment (kristoferlund) [inline packages/api2/src/event-log/event-log.controller.ts:5]: The api enpoints are `/all` and `/types`.

File: packages/api2/src/event-log/entities/event-log.entity.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Prop({ type: String, required: true })
+  description: string;
+}
+
+// eventLogSchema.plugin(mongoosePagination);
+
+export const EventLogSchema = SchemaFactory.createForClass(EventLog);
+
+// export const EventLogModel = model<
```
Comment (kristoferlund) [inline packages/api2/src/event-log/entities/event-log.entity.ts:47]: Pagination needs to be supported

File: packages/api2/src/_shared/shared.module.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { Module } from '@nestjs/common';
+import { SharedService } from './shared.service';
+
+@Module({
+  providers: [SharedService],
+})
+export class SharedModule {}
```
Comment (kristoferlund) [inline packages/api2/src/_shared/shared.module.ts:7]: What is the purpose of the shared module?

**Review by kristoferlund — CHANGES_REQUESTED** [review]
Looks good so far, some things remain for it to be finished though.

File: packages/api2/src/_shared/shared.module.ts (hunk 3 above)
Comment (saucecodee) [reply packages/api2/src/_shared/shared.module.ts:7]: Shared module contains utils, pipes etc that can’t be in a separate module but are needed in multiple modules

**Pull request merged**